cmd/tools/rpcwrappers/main.go

```diff
@@ -0,0 +1,475 @@
+		var err error
+		ctx, cancel := c.createContext(ctx)
+		defer cancel()
+		response, err = client.{{.Method}}(ctx, request, opts...)
+		return err
+	}
+	err = c.executeWithRedirect(ctx, client, op)
+	if err != nil {
+		return nil, err
+	}
+	return response, nil
+}
+`)
+	// TODO: some methods call client.{{.Method}} directly and do not use executeWithRedirect. should we preserve this?
+	// GetDLQReplicationMessages
+	// GetDLQMessages
+	// PurgeDLQMessages
+	// MergeDLQMessages
+}
+
+func generateMatchingClient(w io.Writer, service service) {
+	writeTemplatedCode(w, service, `
+package {{.ServiceName}}
+
+import (
+	"context"
+
+	"{{.ServicePackagePath}}"
+	"google.golang.org/grpc"
+)
+`)
+
+	writeTemplatedMethods(w, service, "client", `
+func (c *clientImpl) {{.Method}}(
+	ctx context.Context,
+	request {{.RequestType}},
+	opts ...grpc.CallOption,
+) ({{.ResponseType}}, error) {
+
+	{{.GetClient}}
+	if err != nil {
+		return nil, err
+	}
+	ctx, cancel := c.createContext(ctx)
+	defer cancel()
+	return client.{{.Method}}(ctx, request, opts...)
+}
+`)
+}
+
+func generateFrontendOrAdminMetricClient(w io.Writer, service service) {
+	writeTemplatedCode(w, service, `
+package {{.ServiceName}}
+
+import (
+	"context"
+
+	"{{.ServicePackagePath}}"
+	"google.golang.org/grpc"
+
+	"go.temporal.io/server/common/metrics"
+)
+`)
+
+	writeTemplatedMethods(w, service, "metricsClient", `
+func (c *metricClient) {{.Method}}(
+	ctx context.Context,
+	request {{.RequestType}},
+	opts ...grpc.CallOption,
+) ({{.ResponseType}}, error) {
+
+	c.metricsClient.IncCounter(metrics.{{.MetricPrefix}}{{.Method}}Scope, metrics.ClientRequests)
+
+	sw := c.metricsClient.StartTimer(metrics.{{.MetricPrefix}}{{.Method}}Scope, metrics.ClientLatency)
+	resp, err := c.client.{{.Method}}(ctx, request, opts...)
+	sw.Stop()
+
+	if err != nil {
+		c.metricsClient.IncCounter(metrics.{{.MetricPrefix}}{{.Method}}Scope, metrics.ClientFailures)
+	}
+	return resp, err
+}
+`)
+}
+
+func generateHistoryOrMatchingMetricClient(w io.Writer, service service) {
+	writeTemplatedCode(w, service, `
+package {{.ServiceName}}
+
+import (
+	"context"
+
+	"{{.ServicePackagePath}}"
+	"google.golang.org/grpc"
+
+	"go.temporal.io/server/common/metrics"
+)
+`)
+
+	writeTemplatedMethods(w, service, "metricsClient", `
+func (c *metricClient) {{.Method}}(
+	ctx context.Context,
+	request {{.RequestType}},
+	opts ...grpc.CallOption,
+) (_ {{.ResponseType}}, retError error) {
+
+	scope, stopwatch := c.startMetricsRecording(metrics.{{.MetricPrefix}}{{.Method}}Scope)
+	defer func() {
+		c.finishMetricsRecording(scope, stopwatch, retError)
+	}()
+
+	return c.client.{{.Method}}(ctx, request, opts...)
+}
+`)
+	// TODO: some history methods did not touch metrics. should we preserve this?
+	// DescribeHistoryHost
+	// RemoveTask
+	// CloseShard
+	// GetShard
+	// RebuildMutableState
+	// DescribeMutableState
+	// TODO: DeleteWorkflowExecution didn't work like the others in history (the code looked
```

Comment:
`DeleteWorkflowExecution` should work the same as other methods i.e. `TerminateWorkflowExecution`.